### PR TITLE
Add ability to select default homepage

### PR DESCRIPTION
**Pull request by partydragen**
Add ability to select default homepage

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -89,8 +89,16 @@
...
+        $homepage = $pages->getPageByURL(Settings::get('home_type'));
+        if ($homepage != null) {
+            $pages->setActivePage($homepage);
+            require(implode(DIRECTORY_SEPARATOR, [ROOT_PATH, 'modules', $homepage['module'], $homepage['file']]));
+        } else {
+            $pages->setActivePage($pages->getPageByURL('/'));
+            require(ROOT_PATH . '/modules/Core/pages/index.php');
+        }
+
+        Settings::get('home_type');
```
Comment (samerton) [inline index.php]: Is this needed?

File: modules/Core/pages/panel/general_settings.php [hunk 2]
```diff
@@ -228,6 +219,32 @@
...
+]];
+
+foreach ($pages->returnPages() as $key => $page) {
+    if (str_contains($key, '/panel/') || str_contains($key, '/queries/') || str_contains($key, '/user/')) {
+        continue;
+    }
+
+    $homepage_pages[] = [
+        'value' => $key,
+        'name' => $key,
```
Comment (samerton) [inline modules/Core/pages/panel/general_settings.php]: Should the name/key be purified before passing to the template

Comment (partydragen) [reply modules/Core/pages/panel/general_settings.php]: Solved

File: modules/Core/pages/panel/general_settings.php [hunk 3]
```diff
@@ -298,7 +313,7 @@
     'ENABLE_AUTO_LANGUAGE' => $language->get('admin', 'enable_auto_language'),
     'AUTO_LANGUAGE_HELP' => $language->get('admin', 'auto_language_help'),
     'REQUIRE_STAFFCP_TFA' => $language->get('admin', 'require_two_factor_for_staffcp'),
-    'REQUIRE_STAFFCP_TFA_VALUE' => Settings::get('require_staffcp_tfa'),
+    'REQUIRE_STAFFCP_TFA_VALUE' => Settings::get('require_staffcp_tfa')
```
Comment (samerton) [inline modules/Core/pages/panel/general_settings.php:316]: Where did the comma go 😢

**Review by samerton — APPROVED** [review]

File: modules/Core/pages/panel/general_settings.php (hunk 3 above)
Comment (partydragen) [reply modules/Core/pages/panel/general_settings.php:316]: the stupid comma lol xD

**Pull request merged**